packages/opentelemetry-remote-sampler-aws-xray/src/remote-sampler.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Sampler, SamplingDecision, SamplingResult } from '@opentelemetry/sdk-trace-base';
+import { SamplingRule } from './remote-sampler.types';
+import axios from 'axios';
+
+const DEFAULT_INTERVAL = 5 * 60 * 1000;// 5 minutes on sampling rules fetch (default polling interval)
+
+
+// IN PROGRESS - SKELETON CLASS 
+export class AWSXRayRemoteSampler implements Sampler {
+    private _pollingInterval: number;
+    private _endpoint: string;
+    private _samplingRulesEndpoint: string;
+
+    constructor(endpoint: string, pollingInterval: number = DEFAULT_INTERVAL) {
+
+        if (pollingInterval <= 0 || !Number.isInteger(pollingInterval)) {
+            throw new TypeError('pollingInterval must be a positive integer');
+        }
+
+        this._pollingInterval = pollingInterval;
+        this._endpoint = endpoint;
+        this._samplingRulesEndpoint = "/GetSamplingRules";
+
+        // execute first get Sampling rules update using polling interval
+        this.getSamplingRules();
+    }
+
+    shouldSample(): SamplingResult {
+        // Implementation to be added
+        return { decision: SamplingDecision.NOT_RECORD };
+    }
+
+    toString(): string {
+        return `AWSXRayRemoteSampler`;
+    }
+
+    getEndpoint(): string {
+        return this._endpoint;
+    }
+
+    getPollingInterval(): number {
+        return this._pollingInterval;
+    }
+
+
+    // fetch sampling rules every polling interval
+    public async getSamplingRules(): Promise<void> {
+        const endpoint = this._endpoint + this._samplingRulesEndpoint;
+
+        setInterval(async () => {
```

Comment:
The x-ray remote sampler requires that the `/GetSamplingRules` endpoint is called every 5 minutes (or user-defined interval), and it is an essential component to ensuring the sampling behaves correctly. This behavior is implemented in a separate worker thread in our [Go](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/8cf7954c58eab45e3fef799742c3e798c2174a59/samplers/aws/xray/internal/client.go) and [DotNet](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1091) implementations, but since JS is a single-threaded language it is preferred to use async calls through intervals or timeouts. 

Our customer experience is centered around [creating sampling rules in the X-Ray console](https://docs.aws.amazon.com/xray/latest/devguide/xray-console-sampling.html#xray-console-config), which will be fetched periodically using the `/GetSamplingRules` endpoint. If we want to stop the sampler, then the rules would be removed from the X-Ray console and the Sampler would be removed from the `TracerProvider`. 

Please correct me if I'm not misunderstood this part, but I believe that the snippet below from our [documentation on initializing components to send traces to X-Ray](https://aws-otel.github.io/docs/getting-started/js-sdk/trace-manual-instr#sending-traces-to-aws-x-ray) would be used to make sure the application exits nicely. 
```
// gracefully shut down the SDK on process exit
    process.on('SIGTERM', () => {
    sdk.shutdown()
      .then(() => console.log('Tracing and Metrics terminated'))
      .catch((error) => console.log('Error terminating tracing and metrics', error))
      .finally(() => process.exit(0));
    });
```